dask/array/backends.py

```diff
@@ -296,3 +299,157 @@ def _nannumel_sparse(x, **kwargs):
+
+        Returns a new array having a specified shape and filled
+        with ones.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def zeros(shape, *, dtype=None, meta=None):
+        """Create an array of zeros
+
+        Returns a new array having a specified shape and filled
+        with zeros.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def empty(shape, *, dtype=None, meta=None):
+        """Create an empty array
+
+        Returns an uninitialized array having a specified shape.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def full(shape, fill_value, *, dtype=None, meta=None):
+        """Create a uniformly filled array
+
+        Returns a new array having a specified shape and filled
+        with fill_value.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def arange(start, /, stop=None, step=1, *, dtype=None, meta=None):
+        """Create an ascending or descending array
+
+        Returns evenly spaced values within the half-open interval
+        ``[start, stop)`` as a one-dimensional array.
+        """
+        raise NotImplementedError
+
+
+class NumpyBackendEntrypoint(ArrayBackendEntrypoint):
+    def __init__(self):
+        # Importing this class will already guarentee
+        # that data-dispatch functions are registered
+        pass
+
+    @property
+    def RandomState(self):
+        return np.random.RandomState
+
+
+class ArrayCreationDispatch(CreationDispatch):
+    def register_backend(self, backend: str, cls_target: DaskBackendEntrypoint):
+        """Register a target class for a specific array-backend label"""
+
+        def wrapper(cls_target):
+            if isinstance(cls_target, ArrayBackendEntrypoint):
+                self._lookup[backend] = cls_target
+            else:
+                raise ValueError(
+                    f"ArrayCreationDispatch only supports "
+                    f"ArrayBackendEntrypoint registration. "
+                    f"Got {cls_target}"
+                )
+            return cls_target
+
+        return wrapper(cls_target)
+
+
+array_creation_dispatch = ArrayCreationDispatch(
+    config_field="array.backend.library",
+    default="numpy",
+    name="array_creation_dispatch",
+)
+
+
+array_creation_dispatch.register_backend("numpy", NumpyBackendEntrypoint())
+
+
+try:
+    import cupy
+
+    class CupyBackendEntrypoint(ArrayBackendEntrypoint):
+        def __init__(self):
+            # Importing this class will already guarentee
+            # that data-dispatch functions are registered
+            pass
+
+        @property
+        def RandomState(self):
+            return cupy.random.RandomState
+
+        @staticmethod
+        def ones(*args, meta=None, **kwargs):
+            meta = cupy.empty(()) if meta is None else meta
+            with config.set({"array.backend.library": "numpy"}):
+                return da.ones(*args, meta=meta, **kwargs)
```

Comment:
I can understand your skepticism here!

The current Dask-Array convention is to use an optional `meta=` argument for backend dispatching. If `meta` is a cupy array, then they want a cupy-backed Dask-Array. If `meta` is a numpy array, then they want a numpy-backed Dask-Array.

When the user sets the backend to "cupy", the `meta=` argument should still take precedence in determining the Dask-Array backend. The target of config-based dispatching is to set defaults that are not already specified in function arguments.

Would you be more comfortable with something like this?:

```python
        @staticmethod
        def ones(*args, meta=None, **kwargs):
            meta = cupy.empty(()) if meta is None else meta
            return w(broadcast_trick(np.ones_like))(*args, meta=meta, **kwargs)
```

Reply:
>The target of config-based dispatching is to set defaults that are not already specified in function arguments.

I suppose this idea highlights a key challenge here. We are dispatching the function entirely, and leaving it up to the backend to account for optional arguments like `meta` and `engine`, that should *probably* take precedence over the configured backend. This feels icky to me, but I don't know if there is a better way to get what we want.

I suppose one alternative is to standardize the `meta` and `engine` arguments for **all** collection-creation functions, and then use the backend config to dispatch the entrypoint function if `meta`/`engine` is not explicitly specified.  That solution would be a lot more work, but I suppose it *could* be accomplished **after** this PR without too much trouble.

Reply:
I think for this PR the current approach is fine (it just looked surprising to me). I guess you need to undo the config so that you terminate the recursion somehow? Otherwise `da.ones` will call back into this same function?

Reply:
>I guess you need to undo the config so that you terminate the recursion somehow?

Exactly right, you need to break the recursion to call back into the same function.